check-log/lib/check-log.go

```diff
@@ -269,14 +285,30 @@ func (opts *logOpts) searchLog(logFile string) (int64, int64, string, error) {
 		return warnNum, critNum, errLines, err
 	}
 
+	if !opts.NoState && oldf != nil {
+		// search old file
+		var (
+			oldWarnNum, oldCritNum int64
+			oldErrLines            string
+		)
+		// ignore readBytes under the premise that the old file will never be updated.
+		oldWarnNum, oldCritNum, _, oldErrLines, err := opts.searchReader(oldr)
+		if err != nil {
+			return oldWarnNum, critNum, errLines, err
+		}
+		warnNum += oldWarnNum
+		critNum += oldCritNum
+		errLines += oldErrLines
```

Comment:
When `--return` option specified, matched lines will be returned.
```
$ go run main.go --check-first -f README.md --pattern ISO --warning-over=1 --return
LOG CRITICAL: 14 warnings, 14 criticals for pattern /ISO/.
[README.md]
* ISO-2022-JP
* ISO-8859-1
* ISO-8859-2
* ISO-8859-3
* ISO-8859-4
* ISO-8859-5
* ISO-8859-6
* ISO-8859-7
* ISO-8859-8
* ISO-8859-10
* ISO-8859-13
* ISO-8859-14
* ISO-8859-15
* ISO-8859-16

exit status 2
```
With this implementation, matched lines in old file and those in new file will be displayed in same manner and users cannot distinguish. Do you have any idea to solve this problem?
(Though I think this p-r's behavior is acceptable)

Reply:
That's a good point.
I can add a prefix like `[old]` and `[path/to/file]`to displayed lines, but check-log will lost backward compatibility if a user's program parses check-logs's output.